tests/aem.pm

```diff
@@ -0,0 +1,242 @@
+        type_string "srkpass\n";
+        wait_serial "Confirm SRK password:" ;
+        type_string "srkpass\n";
+        wait_serial "tpm-setup-done" ;
+    } else {
+        assert_script_run('anti-evil-maid-tpm-setup -z');
+    }
+}
+
+sub install_aem {
+    # edit configuration to use only PCR[13], which should be non-zero even
+    # without DRTM due to being populated by LUKS's header hash
+    assert_script_run('sed -i \'s/SEAL=.*/SEAL="--pcr 13"/\' /etc/anti-evil-maid.conf');
+
+    if (check_var('TEST_AEM_MFA', '1')) {
+        # make sure time is the same or TOTP won't work out
+        assert_script_run("date -s @" . time());
+
+        type_string "anti-evil-maid-install -m /dev/vda1\n";
+
+        # skip warning about installing MFA on internal disk
+        wait_serial "Press <ENTER> to continue...";
+        send_key "ret";
+
+        wait_serial "Please scan the above QR code";
+
+        # capture and prepare TOTP secret
+        my $secret_re = qr/( [A-Z2-7]{4}){8}/;
+        my $secret = wait_serial($secret_re);
+        $secret =~ s/.*($secret_re).*/$1/s;
+        $secret =~ s/ //g;
+        $totp_secret = $secret;
+
+        my $totp = generate_totp($secret);
+        wait_serial 'Code:';
+        type_string "$totp\n";
+
+        my $result = wait_serial qr/TOTP code .+/;
+        # up to three attempts, but if two failed something other than time is off
+        if ($result =~ 'is invalid') {
+            diag "Parsed TOTP secret: $secret";
+            $totp = generate_totp($secret);
+            wait_serial 'Code:';
+            type_string "$totp\n";
+            wait_serial 'TOTP code matches';
+        }
+
+        wait_serial 'Please enter passphrase:';
+        type_string "aemlukspass\n";
+        wait_serial 'Please confirm passphrase:';
+        type_string "aemlukspass\n";
+
+        wait_serial 'Enter any existing passphrase:';
+        type_string "lukspass\n";
+
+        wait_serial 'done';
+    } else {
+        assert_script_run('anti-evil-maid-install /dev/vda1');
+    }
+
+    assert_script_run('echo "really big secret" > /var/lib/anti-evil-maid/aem/secret.txt');
+}
+
+sub handle_aem_startup {
+    if (check_var('TEST_AEM_MFA', '1')) {
+        my $expected_old = generate_totp($totp_secret);
+        my $needle = assert_screen "aem-totp-code", timeout => 60;
+        my $expected_new = generate_totp($totp_secret);
+
+        # accounting for possible delays by matching one of three areas
+        # against up to two codes
+        my $code = undef;
+        my @ocr = @{$needle->{ocr}};
+        for (my $i = 0; $i <= $#ocr; $i++) {
+            my $text = $ocr[$i];
+            # filtering after capturing more than needed to account for offsets
+            # caused by proportional font
+            $text =~ s/.*(\d{6}).*/$1/s;
+            if ($text eq $expected_old || $text eq $expected_new) {
+                $code = $text;
+                last;
+            }
+        }
```

Comment:
`autoinst-log.txt` contains:
```
Can't exec "tesseract": Permission denied at /usr/lib/os-autoinst/ocr.pm line 20.
```
Which led to `$needle->{ocr}` not being initialized.  For some reason I didn't need to update AppArmor configuration to make OCR work, but I guess it might be necessary.

Reply:
Indeed, default apparmor profile doesn't allow tesseract. Fixed.